src/test/scala/dev/kamu/cli/MetadataRepositorySpec.scala

```diff
@@ -53,4 +58,56 @@ class MetadataRepositorySpec extends FunSuite with Matchers with KamuTestBase {
       actual3 shouldEqual Seq(dsC.id, dsB.id, dsA.id)
     }
   }
+
+  test(raw"'kamu add' from HTTP") {
+    withEmptyRepo { kamu =>
+      val testDir = getRandomDir()
+      val serverPort = 8080
+      val hostPort = 9000
```

Comment:
just fyi, to exclude the possibility of port collision you could expose port without assigning it a number and then use `DockerProcess.getHostPort` method to see which port on the host it was mapped to.